src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -51,12 +53,12 @@ private bool IsUserLog<TState>(TState state)
                 Utility.GetStateBoolValue(stateDict, ScriptConstants.LogPropertyIsUserLogKey) == true);
         }
 
-        private string GetEventName<TState>(TState state)
+        private string GetValueFromState<TState>(TState state, string key)
```

Comment:
This seems generally useful. Did you want to move this to the `Utility` class as another overload like `GetStateValueOrDefault(object state)`?